src/main/groovy/com/pszymczyk/consul/infrastructure/SimpleConsulClient.groovy

```diff
@@ -1,17 +1,36 @@
 package com.pszymczyk.consul.infrastructure
 
 import groovyx.net.http.ContentType
+import groovyx.net.http.EncoderRegistry
 import groovyx.net.http.HttpResponseDecorator
 import groovyx.net.http.RESTClient
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
 
 class SimpleConsulClient {
 
-    private static final String NO_LEADER_ELECTED_RESPONSE = "";
+    private static final String NO_LEADER_ELECTED_RESPONSE = ""
 
     private final RESTClient http
+    private final String token
 
-    SimpleConsulClient(String host, int httpPort) {
+    SimpleConsulClient(String host, int httpPort, String token) {
```

Comment:
There are too much responsibilities in class constructor right now. Maybe try to introduce factory pattern + introduce separate component for http parser?

Reply:
Agree, will do